### ISSUE TITLE
internal error: entered unreachable code

### ISSUE THREAD

**Issue by SuperCuber** [rust-lang/rust-analyzer#11812]
**rust-analyzer version**: rust-analyzer b594f9c44 2022-03-21 stable

**rustc version**: rustc 1.58.1 (db9d1b20b 2022-01-20)

**relevant settings**: N/A

#### Cargo.toml
```toml
[package]
name = "testing"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
thiserror = "1"
```

#### main.rs
```rust
#[derive(Debug, thiserror::Error)]
enum MyError {}

fn main() {}
```

### Expected behavior
No diagnostics shown

#### Windows
Error-level diagnostic shown on the whole derive+enum item saying `internal error: reached unreachable code`

Additionally, sometimes it (used to?) show an error about a pipe being broken.

Forgive the terrible formatting, this is how Neovim's LSP outputs stuff.
<details>
  <summary>LSP Log</summary>

```log
[START][2022-03-24 20:16:35] LSP logging initiated
[START][2022-03-24 20:16:58] LSP logging initiated
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"thread 'main' panicked at 'internal error: entered unreachable code', crates\\proc_macro_srv\\src\\abis\\abi_1_56\\proc_macro\\bridge\\mod.rs:256:5\nstack backtrace:\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"thread 'main' panicked at 'internal error: entered unreachable code', crates\\proc_macro_srv\\src\\abis\\abi_1_56\\proc_macro\\bridge\\mod.rs"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	":256:5\nstack backtrace:\n   0:     0x7ff699109d79 - <unknown>\n   1:    "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	" 0x7ff69819b39a - <unknown>\n   2:     0x7ff6991041b8 - <unknown>\n   "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"3:     0x7ff69910d632 - <unknown>\n   4:     0x7ff69910d1c2 - <unknown>\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"   5:     0x7ff69910dcb4 - <unknown>\n   6:     "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"0x7ff69910d84b - <unknown>\n   7:     0x7ff69910afaf - <unknown>\n   8:  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"   0x7ff69910d7cf - <unknown>\n   9:     0x7ff6991d1090 - <unknown>\n  10"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	":     0x7ff6991d0fdc - <unknown>\n  11:     0x7ff698ce9a60 - <unknown>\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  12:     0x7ff698cb9b06 - <unknown>\n  13:     0x7ff698cf2c44 - <unknown>\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  14:     "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"0x7ff8735a0713 - proc_macro::bridge::closure::Closure::call\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\closure.rs:27\n  15:     0x7ff8735a0713 - proc_macro::bridge::client::impl$154::into_iter::closure$0\n                  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"             at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\client.rs:244\n  16:     0x7ff8735a0713 - proc_macro::bridge::client::impl$9::with::closure$0\n      "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                         at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\client.rs:336\n  17:     0x7ff8735a0713 - proc_macro::bridge::client::impl$8::with::closure$0::closure$0\n            "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                   at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\client.rs:293\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  18:     "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"0x7ff8735a0713 - proc_macro::bridge::scoped_cell::ScopedCell::replace\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\scoped_cell.rs:75\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  19:     0x7ff8735a0713 - proc_macro::bridge::client::impl$8::with::closure$0\n                  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"             at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\client.rs:291\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  20:     0x7ff8735a0713 - std::thread::local::LocalKey::try_with\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\std\\src\\thread\\local.rs:399\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  21:     0x7ff8735a0713 - std::thread::local::LocalKey::with\n              "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                 at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\std\\src\\thread\\local.rs:375\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  22:    "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	" 0x7ff8735a0713 - proc_macro::bridge::client::BridgeState::with\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\client.rs:290"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  23:   "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  0x7ff8735a0713 - proc_macro::bridge::Bridge::with\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\client.rs:"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"329\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  24:     0x7ff8735a0713 - proc_macro::bridge::client::TokenStream::into_iter\n  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                             at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\bridge\\client.rs:237\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  25:    "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	" 0x7ff8735a0713 - proc_macro::token_stream::impl$1::into_iter\n            "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                   at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\/library\\proc_macro\\src\\lib.rs:293\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  26: "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"    0x7ff87357e9db - proc_macro2::imp::impl$19::into_iter\n              "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                 at C:\\Users\\Amit\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\proc-macro2-1.0.36\\src\\wrapper.rs:314\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  27:     0x7ff87357bdf0"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	" - proc_macro2::token_stream::impl$2::into_iter\n                  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"             at C:\\Users\\Amit\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\proc-macro2-1.0.36\\src\\lib.rs:1307\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  28:     0x7ff873526068 - syn::buffer::TokenBuffer::inner_new\n "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                              at C:\\Users\\Amit\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\syn-1.0.89\\src\\buffer.rs:64\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  29: "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"    0x7ff873526a32 - syn::buffer::TokenBuffer::new2\n          "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                     at C:\\Users\\Amit\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\syn-1.0.89\\src\\buffer.rs:128\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  30:     0x7ff873480617 - syn::parse::impl$17::parse2<enum$<core::result::Result<syn::derive::DeriveInput,syn::error::Error> > (*)(ref$<syn::parse::ParseBuffer>),syn::derive::DeriveInput>\n                 "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"              at C:\\Users\\Amit\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\syn-1.0.89\\src\\parse.rs:1209\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  31:     0x7ff8734801f0 - syn::parse::Parser::parse<enum$<core::result::Result<syn::derive::DeriveInput,syn::error::Error> > (*)(ref$<syn::parse::ParseBuffer>)>\n            "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                   at C:\\Users\\Amit\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\syn-1.0.89\\src\\parse.rs:1163\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  32:     0x7ff87347fce9 - syn::parse_macro_input::parse<syn::derive::DeriveInput>\n   "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                            at C:\\Users\\Amit\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\syn-1.0.89\\src\\parse_macro_input.rs:139\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  33:     0x7ff8734a230f - thiserror_impl::derive_error\n                               at C:\\Users\\Amit\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\thiserror-impl-1.0.30\\src\\lib.rs:28\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  34:     0x7ff87348131e - core::ops::function::FnOnce::call_once<proc_macro::TokenStream (*)(proc_macro::TokenStream),tuple$<proc_macro::TokenStream> >\n      "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                         at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\core\\src\\ops\\function.rs:227\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  35:    "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	" 0x7ff8734910a6 - proc_macro::bridge::client::impl$10::expand1::run::closure$0<proc_macro::TokenStream (*)(proc_macro::TokenStream)>\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\client.rs:410"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"\n  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"36:     0x7ff873490742 - proc_macro::bridge::client::run_client::closure$0::closure$0<proc_macro::bridge::client::TokenStream,proc_macro::bridge::client::TokenStream,proc_macro::bridge::client::impl$10::expand1::run::closure$0>\n        "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                       at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\client.rs:377\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  37:     0x7ff873476558 - proc_macro::bridge::scoped_cell::impl$3::set::closure$0<enum$<proc_macro::bridge::client::BridgeStateL>,tuple$<>,proc_macro::bridge::client::run_client::closure$0::closure$0>\n  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                             at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\scoped_cell.rs:80\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  38:  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"   0x7ff873476e8c - proc_macro::bridge::scoped_cell::ScopedCell<enum$<proc_macro::bridge::client::BridgeStateL> >::replace<enum$<proc_macro::bridge::client::BridgeStateL>,tuple$<>,proc_macro::bridge::scoped_cell::impl$3::set::closure$0>\n             "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                  at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\scoped_cell.rs:75\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  39:     0x7ff873476537 - proc_macro::bridge::scoped_cell::ScopedCell<enum$<proc_macro::bridge::client::BridgeStateL> >::set<enum$<proc_macro::bridge::client::BridgeStateL>,tuple$<>,proc_macro::bridge::client::run_client::closure$0::closure$0>\n      "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                         at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\scoped_cell.rs:80\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  40:     0x7ff873490e8f - proc_macro::bridge::client::impl$9::enter::closure$1<tuple$<>,proc_macro::bridge::client::run_client::closure$0::closure$0>\n  "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                             at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\client.rs:325\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  41"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	":     0x7ff873477b1a - std::thread::local::LocalKey<proc_macro::bridge::scoped_cell::ScopedCell<enum$<proc_macro::bridge::client::BridgeStateL> > >::try_with<proc_macro::bridge::scoped_cell::ScopedCell<enum$<proc_macro::bridge::client::BridgeStateL> >,proc_macro::bridge::client\n           "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                    at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\std\\src\\thread\\local.rs:399\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  42:     0x7ff873477723 - std::thread::local::LocalKey<proc_macro::bridge::scoped_cell::ScopedCell<enum$<proc_macro::bridge::client::BridgeStateL> > >::with<proc_macro::bridge::scoped_cell::ScopedCell<enum$<proc_macro::bridge::client::BridgeStateL> >,proc_macro::bridge::client::im\n       "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                        at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\std\\src\\thread\\local.rs:375\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  43:     0x7ff87346a277 - proc_macro::bridge::Bridge::enter<tuple$<>,proc_macro::bridge::client::run_client::closure$0::closure$0>\n    "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                           at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\client.rs:325\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  44:     "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"0x7ff8734905b2 - proc_macro::bridge::client::run_client::closure$0<proc_macro::bridge::client::TokenStream,proc_macro::bridge::client::TokenStream,proc_macro::bridge::client::impl$10::expand1::run::closure$0>\n                 "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"              at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\client.rs:370\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  45:     0x7ff87349d531 - core::panic::unwind_safe::impl$23::call_once<tuple$<>,proc_macro::bridge::client::run_client::closure$0>\n              "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                 at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\core\\src\\panic\\unwind_safe.rs:271\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  46:     0x7ff87349d95b - std::panicking::try::do_call<core::panic::unwind_safe::AssertUnwindSafe<proc_macro::bridge::client::run_client::closure$0>,tuple$<> >\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\std\\src\\panicking.rs:406\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  47:     0x7ff8734a2523 - thiserror_impl::derive_error::closure$0\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  48:    "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	" 0x7ff87349d89b - std::panicking::try<tuple$<>,core::panic::unwind_safe::AssertUnwindSafe<proc_macro::bridge::client::run_client::closure$0> >\n                 "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"              at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\std\\src\\panicking.rs:370\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  49:     0x7ff8734913a4 - std::panic::catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<proc_macro::bridge::client::run_client::closure$0>,tuple$<> >\n         "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"                      at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\std\\src\\panic.rs:133\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  50:     0x7ff873490493"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	" - proc_macro::bridge::client::run_client<proc_macro::bridge::client::TokenStream,proc_macro::bridge::client::TokenStream,proc_macro::bridge::client::impl$10::expand1::run::closure$0>\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\client.rs:369\n  51:     0x7ff87349105c - proc_macro::bridge::client::impl$10::expand1::run<proc_macro::TokenStream (*)(proc_macro::TokenStream)>\n                               at /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b\\library\\proc_macro\\src\\bridge\\client.rs:410\n  52:     0x7ff698cb98cd - <unknown>\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  53:     0x7ff698d05c3a - <unknown>\n  54:     0x7ff698ccb784 - <unknown>\n  55:     0x7ff698d0025c - <unknown>\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  56:     0x7ff6990a606b - <unknown>\n  57:     0x7ff6990a4e61 - <unknown>\n  58:     0x7ff69909edb6 - <unknown>\n"
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"  59:     0x7ff6990ba6fc - <unknown>\n  60:     0x7ff69910e104 - <unknown>\n  61:     0x7ff6990a9f97 - <unknown>\n  62:     "
[ERROR][2022-03-24 20:17:13] .../vim/lsp/rpc.lua:417	"rpc"	"rust-analyzer"	"stderr"	"0x7ff6991ca260 - <unknown>\n  63:     0x7ff89d477034 - BaseThreadInitThunk\n  64:     0x7ff89f002651 - RtlUserThreadStart\nthread panicked while panicking. aborting.\n"
```
</details>

#### Linux
A hint-level diagnostic is shown on `thiserror::Error` saying `proc macro 'Error' not expanded`, not remedied by restarting the editor.
Nothing is shown in the log - don't know how to get more info on this one.

### Relevant reading
#10772

**Comment by jonas-schievink** [comment]
We're using `abi_1_56` but you're on Rust 1.58.1, so that explains why there's an error. I'm not sure why the detection is failing though.

**Comment by lnicola** [comment]
Try `cargo clean`, maybe?

**Comment by SuperCuber** [comment]
> Try `cargo clean`, maybe?

No bueno

**Comment by lnicola** [comment]
Did you restart RA afterwards?

**Comment by SuperCuber** [comment]
@lnicola yeah. close editor, delete target dir, open project, error is there. (same stack trace with 1.56)

**Comment by SuperCuber** [comment]
It seems it fixes itself between updates.
